### DISCUSSION TITLE
`Networkx.algorithms.community` doesn't have the attribute of `louvain_communities`

### DISCUSSION THREAD

**Discussion by HongduanTian** [networkx/networkx#5885] [Q&A] [answered]
Hi,

When I imported `networkx.algorithms.community` and tried to use `louvain_communities` function, an error was thrown and said `AttributeError:  'networkx.algorithms.community' doesn't have the attribute 'louvain_communities'`.

So, why this error happened? I found the function in this repository.

Best,
hongduan

**Comment by dschult** [comment]
Can you show us the code that you used -- just on a simple graph like `G=nx.path_graph(5)`.
I can't reproduce this error.   
```python
import networkx as nx
import networkx.community as comm   # or import networkx.algorithms.community as comm
comm.louvain_communities(G)
```

**Reply by HongduanTian** [reply]
Hi, thank you for your reply. 

My codes are like this:
`import networkx as nx`
`import networkx.algorithms.community as nxcom`

`# here I generate a graph from a dict, which the keys are node_indices and the values are vectors that contain the similarities with other nodes`
`G = nx.from_dict_of_lists()`
`best_partitions = nxcom.louvain_communities(G, seed=42)`

**Comment by dschult** [comment, marked as answer]
That code works for me.  What version of networkx are you using? `louvain_communities` was added in v2.7.
`print(nx.__version__)`

**Reply by HongduanTian** [reply]
Hi,

The networkx I am using in my env is 2.6.3.

I have tried to upgrade my networkx with `pip install --upgrade networkx`, but it seemed that it didn't work.

**Reply by MridulS** [reply]
That probably means you have python 3.7 on your machine. Could you check your python version with 'python -V' ?

NetworkX 2.7 requires Python 3.8

**Reply by HongduanTian** [reply]
Yes, my python version is 3.7. I will upgrade my python and then try again. Thanks a lot!
